presto-elasticsearch/src/main/java/com/facebook/presto/elasticsearch/ElasticsearchRecordCursor.java

```diff
@@ -0,0 +1,237 @@
+
+        extractFromSource(hit);
+        totalBytes += fields.size();
+        return true;
+    }
+
+    @Override
+    public boolean getBoolean(int field)
+    {
+        checkFieldType(field, ImmutableSet.of(BOOLEAN));
+        return (Boolean) getFieldValue(field);
+    }
+
+    @Override
+    public long getLong(int field)
+    {
+        checkFieldType(field, ImmutableSet.of(BIGINT, INTEGER));
+        return Long.valueOf(String.valueOf(getFieldValue(field)));
+    }
+
+    @Override
+    public double getDouble(int field)
+    {
+        checkFieldType(field, ImmutableSet.of(DOUBLE));
+        return (Double) getFieldValue(field);
+    }
+
+    @Override
+    public Slice getSlice(int field)
+    {
+        checkFieldType(field, ImmutableSet.of(VARCHAR));
+
+        Object value = getFieldValue(field);
+        ElasticsearchColumnHandle column = columnHandles.get(field);
+        if (tupleDomain.getDomains().isPresent()) {
+            Map<ColumnHandle, Domain> domains = tupleDomain.getDomains().get();
+            Domain domain = domains.get(column);
+            if (domain != null) {
+                for (Range range : domain.getValues().getRanges().getOrderedRanges()) {
+                    if (range.isSingleValue()) {
+                        return Slices.utf8Slice(String.valueOf(getValue(column.getColumnType(), range.getLow().getValue())));
+                    }
+                }
+            }
+        }
+        if (value instanceof Collection) {
+            return Slices.utf8Slice(new JSONArray((List<Map<String, Object>>) value).toString());
+        }
+        else {
+            return Slices.utf8Slice(String.valueOf(value));
+        }
+    }
+
+    @Override
+    public Object getObject(int field)
+    {
+        return null;
+    }
+
+    @Override
+    public boolean isNull(int field)
+    {
+        checkArgument(field < columnHandles.size(), "Invalid field index");
+        return getFieldValue(field) == null;
+    }
+
+    private void checkFieldType(int field, Set<Type> expectedTypes)
+    {
+        Type actual = getType(field);
+        StringBuilder builder = new StringBuilder();
+        for (Type type : expectedTypes) {
+            builder.append(", ");
+            builder.append(type.getDisplayName());
+        }
+        checkArgument(expectedTypes.contains(actual), "Expected field %s to be type %s but is %s", field, builder.toString(), actual);
+    }
+
+    @Override
+    public void close()
+    {
+    }
+
+    private List<SearchHit> sendElasticQuery(ElasticsearchQueryBuilder queryBuilder)
+    {
+        List<SearchHit> result = new ArrayList<>();
+        SearchResponse response = queryBuilder.buildScrollSearchRequest().execute().actionGet();
+
+        if (response.getHits().getTotalHits() > maxHits) {
+            throw new PrestoException(ELASTIC_SEARCH_EXCEEDS_MAX_HIT_ERROR, "Query get: " + response.getHits().getTotalHits() + " hits, exceeds Elasticsearch max hits configuration: " + maxHits);
```

Comment:
`format("The number of hits for the query exceeds the configured max hits: %d > %d", response.getHits().getTotalHits(), maxHits)`